mirdata/datasets/dagstuhl_choirset.py

```diff
@@ -0,0 +1,465 @@
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+
+        f0_data = [
+            (self.f0_crepe_dyn, "crepe - DYN"),
+            (self.f0_crepe_hsm, "crepe - HSM"),
+            (self.f0_crepe_lrx, "crepe - LRX"),
+            (self.f0_pyin_dyn, "pyin - DYN"),
+            (self.f0_pyin_hsm, "pyin - HSM"),
+            (self.f0_pyin_lrx, "pyin - LRX"),
+            (self.f0_manual_lrx, "manual - LRX"),
+        ]
+        # remove missing annotations from the list
+        f0_data = [tup for tup in f0_data if tup[1]]
+        score_data = [(self.score, "score")] if self.score else None
+
+        if self.audio_hsm_path:
+            audio_path = self.audio_hsm_path
+        elif self.audio_dyn_path:
+            audio_path = self.audio_dyn_path
+        else:
+            audio_path = self.audio_lrx_path
+
+        return jams_utils.jams_converter(
+            audio_path=audio_path,
+            f0_data=f0_data,
+            note_data=score_data,
+        )
+
+
+class MultiTrack(core.MultiTrack):
+    """Dagstuhl ChoirSet multitrack class
+
+    Args:
+        mtrack_id (str): multitrack id
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/dagstuhl_choirset`
+
+    Attributes:
+        audio_stm_path (str): path to room mic (mono mixdown) audio file
+        audio_str_path (str): path to room mic (right channel) audio file
+        audio_stl_path (str): path to room mic (left channel) audio file
+        audio_rev_path (str): path to room mic with artifical reverb (mono mixdown) audio file
+        audio_spl_path (str): path to piano accompaniment (left channel) audio file
+        audio_spr_path (str): path to piano accompaniement (right channel) audio file
+        beat_path (str): path to beat annotation file
+
+    Cached Properties:
+        beat (annotations.BeatData): Beat annotation
+
+    """
+
+    def __init__(
+        self, mtrack_id, data_home, dataset_name, index, track_class, metadata
+    ):
+
+        super().__init__(
+            mtrack_id=mtrack_id,
+            data_home=data_home,
+            dataset_name=dataset_name,
+            index=index,
+            track_class=track_class,
+            metadata=metadata,
+        )
+
+        self.audio_stm_path = self.get_path("audio_stm")
+        self.audio_str_path = self.get_path("audio_str")
+        self.audio_stl_path = self.get_path("audio_stl")
+        self.audio_rev_path = self.get_path("audio_rev")
+        self.audio_spl_path = self.get_path("audio_spl")
+        self.audio_spr_path = self.get_path("audio_spr")
+        self.beat_path = self.get_path("beat")
+
+    @property
+    def track_audio_property(self):
+        return "audio_dyn"
+
+    @core.cached_property
+    def beat(self) -> Optional[annotations.BeatData]:
+        return load_beat(self.beat_path)
+
+    @property
+    def audio_stm(self) -> Tuple[np.ndarray, float]:
```

Comment:
(sorry for my mistake!) the fix:
`Tuple[np.ndarray, float]` should be `Optional[Tuple[np.ndarray, float]]`